### PR TITLE
Use shared files for version and build numbers, to keep Android and iOS builds in sync.

### PR DESCRIPTION
**Pull request by halmueller**
This is working great on iOS. I have no way to test it on Android, though.

Please pull the branch, and test building for Android. I might have botched the changes I made to `build.gradle`, so if something's wrong, look there.

If it works correctly, please comment. I will update documentation before merging this PR.

### CODE REVIEW

**Review by andaniel2029 — COMMENTED** [review]
Hi, @halmueller 

I tried to build it on my mac machine.

But getting errors like this.

Screen Shot 2021-12-15 at 17 37 59

**Comment by halmueller** [comment]
I don't have any way to debug this. Can you dig in a bit and see what's going on? 

The only Android-side file I touched was `build.gradle`. And I don't know Gradle. Could lines 127-128 require a `def`?

I'm trying to get Gradle to read the two external versioning files instead of hardcoding versioning info in `build.gradle`.

**Review by andaniel2029 — COMMENTED** [review]
Hi, @halmueller 

Here are error messages I faced.

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Volumes/Work/Github/ReactNative/iSeaTree-React-Prototype/android/app/build.gradle' line: 130

* What went wrong:
A problem occurred evaluating project ':app'.
> No signature of method: build_5kxrtbi3opjer0qddkg6nuszp.android() is applicable for argument types: (build_5kxrtbi3opjer0qddkg6nuszp$_run_closure1) values: [build_5kxrtbi3opjer0qddkg6nuszp$_run_closure1@6e3ec573]

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```

**Comment by halmueller** [comment]
If we are going to get this working, I need you to dive in and fix the Android side. I cannot install an Android build system on any of  my machines right now.

Do you see what I'm trying to do in `build.gradle`? Load version and build numbers from external file? Please look at the diff of that file in the PR; I'm sure it's a minor thing that I got wrong.

I believe https://stackoverflow.com/questions/35924721/how-to-update-version-number-of-react-native-app will be helpful; that's what I was working from, among other things I found. See also https://discuss.gradle.org/t/a-programatic-way-to-set-the-version-of-any-gradle-project/23197/2 and https://github.com/uklance/tiny-ioc/blob/master/build.gradle#L12, both of which are quite old.

**Comment by andaniel2029** [comment]
> If we are going to get this working, I need you to dive in and fix the Android side. I cannot install an Android build system on any of my machines right now.
> 
> Do you see what I'm trying to do in `build.gradle`? Load version and build numbers from external file? Please look at the diff of that file in the PR; I'm sure it's a minor thing that I got wrong.
> 
> I believe https://stackoverflow.com/questions/35924721/how-to-update-version-number-of-react-native-app will be helpful; that's what I was working from, among other things I found. See also https://discuss.gradle.org/t/a-programatic-way-to-set-the-version-of-any-gradle-project/23197/2 and https://github.com/uklance/tiny-ioc/blob/master/build.gradle#L12, both of which are quite old.

ok, let me see.

**Comment by andaniel2029** [comment]
> If we are going to get this working, I need you to dive in and fix the Android side. I cannot install an Android build system on any of my machines right now.
> 
> Do you see what I'm trying to do in `build.gradle`? Load version and build numbers from external file? Please look at the diff of that file in the PR; I'm sure it's a minor thing that I got wrong.
> 
> I believe https://stackoverflow.com/questions/35924721/how-to-update-version-number-of-react-native-app will be helpful; that's what I was working from, among other things I found. See also https://discuss.gradle.org/t/a-programatic-way-to-set-the-version-of-any-gradle-project/23197/2 and https://github.com/uklance/tiny-ioc/blob/master/build.gradle#L12, both of which are quite old.

Hi, @halmueller 

For android build version number and name, I think it will be better to put it on the root build Gradle file like other config values.

image

**Comment by halmueller** [comment]
That looks great, thank you! I would never have found that myself.

I'll update instructions and then merge this in.

**Comment by andaniel2029** [comment]
> That looks great, thank you! I would never have found that myself.
> 
> I'll update instructions and then merge this in.

ok, cool.

File: android/build.gradle [hunk 1]
```diff
@@ -6,6 +6,8 @@ buildscript {
         minSdkVersion = 21
         compileSdkVersion = 29
         targetSdkVersion = 30
+        versionCode = 56
+        versionName = "2.2.1"
```
Comment (halmueller) [inline android/build.gradle]: I have corrected this, I think. We specifically DO NOT WANT these numbers hardcode in the file. The whole point of this branch is to have ONE SINGLE PLACE to edit to change both iOS and Android versions.

**Pull request merged**